sync/src/synchronizer/mod.rs

```diff
@@ -207,21 +213,20 @@ impl Synchronizer {
     pub fn eviction(&self, nc: &dyn CKBProtocolContext) {
         let mut peer_states = self.peers().state.write();
         let active_chain = self.shared.active_chain();
-        let is_initial_header_sync = self.shared.state().is_initial_header_sync();
         let mut eviction = Vec::new();
+        let better_tip = self.better_tip();
         for (peer, state) in peer_states.iter_mut() {
             let now = unix_time_as_millis();
 
-            // headers_sync_timeout
-            if let Some(timeout) = state.headers_sync_timeout {
-                if is_initial_header_sync {
-                    if now > timeout && !state.disconnect {
+            if let Some(ref mut process) = state.headers_sync_process {
+                if let Some(is_timeout) = process.is_timeout(&better_tip, now) {
+                    if is_timeout && !state.disconnect {
                         eviction.push(*peer);
                         state.disconnect = true;
                         continue;
                     }
                 } else {
-                    state.headers_sync_timeout = None
+                    active_chain.send_getheaders_to_peer(nc, *peer, &better_tip);
```

Comment:
This is duplicated with the feature `CHAIN_SYNC_TIMEOUT`. Maybe we can merge these two features into one. Or reimplemented these two features, one focuses on the headers sync and another handles the active chain tip.

Reply:
I should remove the timeout of header sync when IBD was done. Then both features are independent.

I will update this.